client.js

```diff
@@ -41,11 +47,19 @@ const Empirica = {
   },
 
   introSteps(func) {
-    config.introSteps = func;
+    if (!_.isFunction(func)) {
+      throw new Error("introSteps requires a function");
+    }
+
+    config.introSteps = isReactComponents(func()) ? func : () => [];
```

Comment:
This will not work. The function passed to `Empirica.introSteps` expects to receive game and treatment as arguments. Here, you're calling the function without these arguments (because at this point, you do not have those objects, we're before any game has started). The check must be done at runtime, not init as you do here.

```js
    config.introSteps = function() {
      const results = func.apply(null, arguments);
      if (isReactComponents(results)) {
        return results;
      }   else {
        console.error("Empirica.introSteps() is not returning an array of components as expected.");
        return [];
      }
   }
```

This is the same for exist steps.

Reply:
Actually, we should also support null/undefined results here. So "maybe" something more like this:

```js
    config.introSteps = function() {
      const results = func.apply(null, arguments);
      if (!results || isReactComponents(results)) {
        return results || [];
      }   else {
        console.error("Empirica.introSteps() is not returning an array of components as expected.");
        return [];
      }
    }
```